### PR TITLE
Remove obsolete code from `utils/console.py`

### PR DESCRIPTION
**Pull request by eerovaher**
### Description

I did not go through the file very thoroughly, but I still managed to find enough obsolete code for a pull request.

- [x] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

File: astropy/utils/console.py [hunk 1]
```diff
@@ -97,28 +88,6 @@ def ipyio(cls):
...
-        return getattr(cls.ipyio, stream)
-
-
-def _get_stdout(stderr=False):
-    """
-    This utility function contains the logic to determine what streams to use
-    by default for standard out/err.
-
-    Typically this will just return `sys.stdout`, but it contains additional
-    logic for use in IPython on Windows to determine the correct stream to use
```
Comment (pllim) [inline astropy/utils/console.py:111]: tl;dr -- Is "IPython on Windows" also not a problem anymore? Not obvious if this is tied to IPython 4 or not.

File: astropy/utils/console.py [hunk 2]
```diff
@@ -359,13 +328,6 @@ def color_print(*args, end="\n", **kwargs):
         write(end)
 
 
-def strip_ansi_codes(s):
```
Comment (pllim) [inline astropy/utils/console.py:362]: Hmm I wonder what this was used for originally but looks okay to remove.

**Review by pllim — COMMENTED** [review]
Thanks!

File: astropy/utils/console.py (hunk 1 above)
Comment (eerovaher) [reply astropy/utils/console.py:111]: Apparently you didn't think this was a problem when you removed all the Ipython-related code from this function in 5ad0a5c83e6dafdc6dbdbef8c52137383b3abccf.

Comment (pllim) [reply astropy/utils/console.py:111]: I cannot tell you what I was thinking anymore over 3 years ago in #10942 . Looks like I was just trying get rid of deprecation warning but this was not affected. Hmm.

**Comment by pllim** [comment]
Failures are unrelated but extra pairs of eyes would help.

File: astropy/utils/console.py (hunk 2 above)
Comment (eerovaher) [reply astropy/utils/console.py:362]: This function was introduced in cb9e50d027066238aa3a98c03dddb05ac1e5a65b (AuthorDate: Mon Dec 5 21:03:47 2011 -0500) and used in a test:
```
$  git grep --files-with-matches strip_ansi_codes cb9e50d027066238aa3a98c03dddb05ac1e5a65b
cb9e50d027066238aa3a98c03dddb05ac1e5a65b:astropy/io/vo/tests/vo_test.py
cb9e50d027066238aa3a98c03dddb05ac1e5a65b:astropy/utils/console.py
```
All uses of it were removed in 46576785b98a43945c753f336b4eb6ecb844b510 (AuthorDate: Tue Dec 6 11:12:41 2011 -0500):
```
$ git grep --count strip_ansi_codes 46576785b98a43945c753f336b4eb6ecb844b510
46576785b98a43945c753f336b4eb6ecb844b510:astropy/utils/console.py:1
```

Comment (pllim) [reply astropy/utils/console.py:362]: Wow, back in #101 . Thanks for checking!

**Review by mhvk — APPROVED** [review]
Looks good to me -- I agree that I'm not 100% sure about everything else, but the changes themselves should have no effect and simplify things nicely. Thanks, @eerovaher!

**Review by neutrinoceros — COMMENTED** [review]
I independently found another block of dead code in this file today. I can open a separate PR for it but I'm happy to see it removed in the present PR too:
```patch
diff --git a/astropy/utils/console.py b/astropy/utils/console.py
index b758122821..9785709b47 100644
--- a/astropy/utils/console.py
+++ b/astropy/utils/console.py
@@ -146,19 +146,6 @@ def isatty(file):
     if getattr(file, "name", None) == "stdout":
         return True
 
-    if hasattr(file, "stream"):
-        # FIXME: pyreadline has no had new release since 2015, drop it when
-        #        IPython minversion is 5.x.
-        # On Windows, in IPython 2 the standard I/O streams will wrap
-        # pyreadline.Console objects if pyreadline is available; this should
-        # be considered a TTY.
-        try:
-            from pyreadline.console import Console as PyreadlineConsole
-        except ImportError:
-            return False
-
-        return isinstance(file.stream, PyreadlineConsole)
-
     return False
 ```

**Comment by neutrinoceros** [comment]
Also, as mentioned by @pllim , failures in extra CI are indeed unrelated. I'll link the issue too: #16016 (thank you @pllim for reporting it !)

**Comment by eerovaher** [comment]
The oldest version of `ipython` we claim to support is still older than 5.0: https://github.com/astropy/astropy/blob/b299eb52c0249df7c0ab4be1447e20ea2aead0cc/pyproject.toml#L89
Therefore the code @neutrinoceros highlighted can't be removed yet. [SPEC 0](https://scientific-python.org/specs/spec-0000/) says we don't have to support `ipython` versions older than 7.32, so bumping our `ipython` requirement would not be a bad idea, but it would deserve a separate pull request.

**Comment by neutrinoceros** [comment]
I see, my bad ! I consulted SPEC0 for reference but missed that our current requirement is way older, thanks for double checking !

**Comment by pllim** [comment]
> it would deserve a separate pull request

Indeed.

Since we have a few extra eyes looked at this and approved. I am happy to merge but would be good to know if @eerovaher wants to incorporate https://github.com/astropy/astropy/pull/16012#pullrequestreview-1872149843 in this PR or save it for follow-up. Thanks, all!

**Comment by eerovaher** [comment]
We cannot remove the code highlighted in https://github.com/astropy/astropy/pull/16012#pullrequestreview-1872149843 until we bump our `ipython` requirement and I thought we all agreed that we shouldn't do that in this pull request.

**Review by pllim — APPROVED** [review]
Okay, let's merge then. Thanks for the clean-up!

**Pull request merged**